### ISSUE TITLE
 opencv-0.93.0/build.rs:203:26 compile error:expected `&str`, found `String`

### ISSUE THREAD

**Issue by phial3** [twistedfall/opencv-rust#620]
Before reporting an issue please first check the
[troubleshooting guide](https://github.com/twistedfall/opencv-rust/blob/master/README.md#troubleshooting). If
the issue you're encountering is not solved thereby please state the following in your bugreport:
1. Operating system : MacOS M2 
2. The way you installed OpenCV: package by homebrew
3. OpenCV version : 4.10.0
4. rustc version : rustc 1.83.0-nightly (1d68e6dd1 2024-09-21)
5. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
   ```

```text
error[E0308]: mismatched types
   --> /Users/admin/.cargo/registry/src/rsproxy-0dccff568467c15b/opencv-0.93.0/build.rs:203:26
    |
203 |             out.flag_if_supported(format!("-F{}", p.to_str().expect("Can't convert path to str")));
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found `String`
    |
    = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by twistedfall** [comment]
That's weird looks like you have an old `cc` crate, can you please show the output you're getting when running in the project directory:
```
cargo tree | grep cc
```

**Comment by twistedfall** [comment]
Although I must say that no version of `cc` after the minimum supported 1.0.83 has had `&str` as an argument to `flag_if_supported()`, they all have `impl AsRef<OsStr>`. Maybe it's a regression in rustc's nightly?

**Comment by phial3** [comment]
> That's weird looks like you have an old `cc` crate, can you please show the output you're getting when running in the project directory:
> 
> ```
> cargo tree | grep cc
> ```

this is cc v1.0.83

**Comment by phial3** [comment]
> > That's weird looks like you have an old `cc` crate, can you please show the output you're getting when running in the project directory:
> > ```
> > cargo tree | grep cc
> > ```
> 
> this is cc v1.0.83

I don't know why is that, rust stable 1.81.0 is the same mistake.

use `opencv = "0.92"`, it's good

**Comment by twistedfall** [comment]
Can you please also try doing `cargo clean` to see if it helps with the issue?

**Comment by phial3** [comment]
> Can you please also try doing `cargo clean` to see if it helps with the issue?

I've tried it, and the mistake is still there

```text
error[E0308]: mismatched types
   --> /Users/admin/.cargo/registry/src/rsproxy-0dccff568467c15b/opencv-0.93.0/build.rs:203:26
    |
203 |             out.flag_if_supported(format!("-F{}", p.to_str().expect("Can't convert path to str")));
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found `String`
    |
    = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
error: could not compile `opencv` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...

```

**Comment by twistedfall** [comment]
Thank you for reporting! This regression should now be fixed in v0.93.1

**Comment by phial3** [comment]
> Thank you for reporting! This regression should now be fixed in v0.93.1

👍👍👍
